presto-docs/src/main/sphinx/admin/properties.rst

```diff
@@ -58,6 +60,251 @@ General Properties
+    * **Default value:** ``JVM max memory`` * ``0.1``
+
+    The purpose of that is same as of :ref:`query.max-memory<query-max-memory>` but the memory is not counted
+    cluster-wise but node-wise instead. This should not be any lower than ``query.max-memory / number of nodes``.
+    It may be required to increase this value if data are skewed.
+
+
+.. _query-max-run-time:
+
+``query.max-run-time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``duration``
+    * **Default value:** ``100d``
+
+    Used as default for session property ``query_max_run_time``. Queries that last longer than the duration set with
+    ``query.max-run-time`` will be cancelled. It is important to set this value to much
+    higher than the average query time to avoid problems with outliers (some queries may randomly take much longer due
+    to cluster load and other circumstances). If Presto runs in an environment where there are mostly very short queries
+    then it may be a good idea to set it to a small value to detect user errors in queries.
+    It may also be decreased in poor Presto cluster configuration with mostly short queries to increase garbage
+    collection efficiency and by that lowering memory usage in cluster. As the query timed out by this limit immediately
+    returns all used resources this may be particularly useful in query management systems to force user limits. The
+    session property ``query_max_run_time`` may also be set to a value less than ``query.max-run-time`` in order to
+    crosscheck for bugs in the query.
+
+
+``query.max-cpu-time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``duration``
+    * **Minimum Value:** ``1ns``
+    * **Default value:** ``1000000000d``
+
+    Used as default for session property ``query_max_cpu_time``. Similar to ``query.max-run-time`` but this sets a
+    maximum CPU time limit of a query. Queries whose CPU time hits a value more than the duration set with
+    ``query.max-cpu-time`` will be cancelled.
+
+
+``query.max-execution-time``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``duration``
+    * **Default value:** ``100d``
+
+    Used as default for session property ``query_max_execution_time``. Unlike ``query.max-run-time`` which enforces
+    starting from the query creation time, this property does not consider the queueing time.
+
+
+``query.queue-config-file``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Default value:**
+
+    **Deprecated** The path to the queue config file. Queues are used to manage the number of concurrent queries across
+    the system. More information on queues and how to configure them can be found in :doc:/admin/queue.
+
+
+``query.remote-task.max-callback-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``integer``
+    * **Minimum Value:** ``1``
+    * **Default value:** ``1000``
+
+    This value describes the maximum size of the thread pool used to handle responses to HTTP requests for
```

Comment:
`handle responses to HTTP requests` -> `run callbacks generated while processing HTTP requests`?